metals/src/main/scala/scala/meta/internal/metals/ScalacDiagnostic.scala

```diff
@@ -3,11 +3,30 @@ package scala.meta.internal.metals
 import org.eclipse.{lsp4j => l}
 
 object ScalacDiagnostic {
+
   object SymbolNotFound {
     private val regex = """not found: (value|type) (\w+)""".r
     def unapply(d: l.Diagnostic): Option[String] = d.getMessage() match {
       case regex(_, name) => Some(name)
       case _ => None
     }
   }
+  object MissingImplementation {
+    // https://github.com/scala/scala/blob/4c0f49c7de6ba48f2b0ae59e64ea94fabd82b4a7/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala#L566
+    private val regex = """needs to be abstract""".r
+    def unapply(d: l.Diagnostic): Option[String] =
+      regex.findFirstMatchIn(d.getMessage()) match {
+        case Some(_) => Some(d.getMessage())
+        case _ => None
+      }
```

Comment:
nit: this could be (for consistency)

```scala
private val regex = """.*needs to be abstract.*""".r
def unapply(d: l.Diagnostic): Option[String] = d.getMessage() match {
  case regex() => Some(d.getMessage())
  case _ => None
}
```

Reply:
done https://github.com/scalameta/metals/pull/1379/commits/15d8e1feba9aed6cce18d9f6b74382f464b8be1e 👍 

Added `(?s)` to enable `.` to capture newlines (because the diagnostic usually have multiple lines).
https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#DOTALL